Src/DependencyCollector/Shared/Implementation/TelemetryDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,293 @@
+            {
+                DependencyCollectorEventSource.Log.TelemetryDiagnosticSourceListenerFailedToSubscribe(ex.ToInvariantString());
+            }
+        }
+
+        /// <summary>
+        /// This method gets called once for each existing DiagnosticListener when this
+        /// DiagnosticListener is added to the list of DiagnosticListeners
+        /// (<see cref="System.Diagnostics.DiagnosticListener.AllListeners"/>). This method will
+        /// also be called for each subsequent DiagnosticListener that is added to the list of
+        /// DiagnosticListeners.
+        /// <seealso cref="IObserver{T}.OnNext(T)"/>
+        /// </summary>
+        /// <param name="value">The DiagnosticListener that exists when this listener was added to
+        /// the list, or a DiagnosticListener that got added after this listener was added.</param>
+        public void OnNext(DiagnosticListener value)
+        {
+            if (value != null)
+            {
+                IDisposable subscription = value.Subscribe(
+                    new IndividualDiagnosticSourceListener(value, this),
+                    (evnt, r, _) => evnt.EndsWith(ActivityNameSuffix, StringComparison.OrdinalIgnoreCase)
+                        || evnt.EndsWith(ActivityStopNameSuffix, StringComparison.OrdinalIgnoreCase));
+
+                if (this.individualSubscriptions == null)
+                {
+                    this.individualSubscriptions = new List<IDisposable>();
+                }
+                this.individualSubscriptions.Add(subscription);
+            }
+        }
+
+        /// <summary>
+        /// Notifies the observer that the provider has finished sending push-based notifications.
+        /// <seealso cref="IObserver{T}.OnCompleted()"/>
+        /// </summary>
+        public void OnCompleted()
+        {
+        }
+
+        /// <summary>
+        /// Notifies the observer that the provider has experienced an error condition.
+        /// <seealso cref="IObserver{T}.OnError(Exception)"/>
+        /// </summary>
+        /// <param name="error">An object that provides additional information about the error.</param>
+        public void OnError(Exception error)
+        {
+        }
+
+        public void Dispose()
+        {
+            if (this.individualSubscriptions != null)
+            {
+                foreach (var individualSubscription in this.individualSubscriptions)
+                {
+                    individualSubscription.Dispose();
+                }
+            }
+
+            if (this.listenerSubscription != null)
+            {
+                this.listenerSubscription.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Handler for Activity stop event (response is received for the outgoing request).
+        /// </summary>
+        /// <param name="diagnosticListener">The diagnostic source of the activity.</param>
+        internal void OnActivityStop(DiagnosticListener diagnosticListener)
```

Comment:
In case I want to implement http-specific listener, would have to implement another subscriber and listener.
We can, of course, refactor it later, but it's rarely a priority and after a while, it becomes really complicated. It would be great if to design it right now to be extendable.

Reply:
I'll see if I can reasonably combine them together, but I don't think this should be a blocker to get this into 2.5

Reply:
Oh, is there a plan to make it into 2.5?

What about lib instrumentation and guidance? Should it also be done in 2.5 timeframe?

Reply:
guidence - yes, I'll start working on it this week and add it to this PR
lib instrumentation is independent (as long as it follows guidance) - we've already started working on 2 but it's not clear when they will be done as they have different dev cycle